pkg/standalone-utils/test/SiloWrapping.test.ts

```diff
@@ -0,0 +1,811 @@
+
+import Token from '@balancer-labs/v2-helpers/src/models/tokens/Token';
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+import StablePool from '@balancer-labs/v2-helpers/src/models/pools/stable/StablePool';
+import { StablePoolEncoder, SwapKind } from '@balancer-labs/balancer-js';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { expectTransferEvent } from '@balancer-labs/v2-helpers/src/test/expectTransfer';
+import { deploy, deployedAt } from '@balancer-labs/v2-helpers/src/contract';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+import { ANY_ADDRESS, MAX_INT256, MAX_UINT256, ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+import { BigNumberish, fp } from '@balancer-labs/v2-helpers/src/numbers';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { Account } from '@balancer-labs/v2-helpers/src/models/types/types';
+import TypesConverter from '@balancer-labs/v2-helpers/src/models/types/TypesConverter';
+import { Dictionary } from 'lodash';
+import { expectChainedReferenceContents, toChainedReference } from './helpers/chainedReferences';
+
+describe('SiloWrapping', function () {
+  let DAI: Token, sDAI: Token, mockSilo: Contract;
+  let senderUser: SignerWithAddress, recipientUser: SignerWithAddress, admin: SignerWithAddress;
+  let vault: Vault;
+  let relayer: Contract, relayerLibrary: Contract;
+
+  before('setup signer', async () => {
+    [, admin, senderUser, recipientUser] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy Vault', async () => {
+    vault = await Vault.create({ admin });
+    // Set up and deploy tokens
+    DAI = await deploy('v2-solidity-utils/TestToken', {
+      args: ['DAI', 'DAI', 18],
+    });
+
+    mockSilo = await deploy('MockSilo', {
+      args: [DAI.address],
+    });
+
+    sDAI = await deploy('MockShareToken', {
+      args: ['sDAI', 'sDAI', 18, mockSilo.address, DAI.address],
+    });
+
+    // initalize the asset storage mapping within the Silo for the main token
+    await mockSilo.setAssetStorage(
+      DAI.address, // interestBearingAsset
+      sDAI.address, // CollateralToken
+      sDAI.address, // CollateralOnlyToken (using wrapped token as a placeholder)
+      sDAI.address, // debtToken (using wrapped token as a placeholder)
+      fp(20000), // totalDeposits; These values do not matter for the sack of relayer tests
+      fp(100), // collateralOnlyDeposits; These values do not matter for the sack of relayer tests
+      fp(9000) // totalBorrowAmount; These values do not matter for the sack of relayer tests
+    );
+  });
+
+  sharedBeforeEach('mint tokens to senderUser', async () => {
+    await DAI.mint(senderUser.address, fp(100));
+    await sDAI.mint(senderUser.address, fp(100));
+
+    await DAI.connect(senderUser).approve(vault.address, fp(100));
+    await sDAI.connect(senderUser).approve(sDAI.address, fp(100));
```

Comment:
We don't need this self-approval, and if we approve the Vault instead then we can avoid approving it later.
```suggestion
    await sDAI.connect(senderUser).approve(vault.address, fp(100));
```